src/mechanics/assembly_elasticity.hh

```diff
@@ -0,0 +1,695 @@
+    shared_ptr<FiniteElement<dim-1>> fe_low_;   ///< Finite element for the solution of the advection-diffusion equation (dim-1).
+
+    /// Data objects shared with Elasticity
+    EqFields *eq_fields_;
+    EqData *eq_data_;
+
+    /// Sub field set contains fields used in calculation.
+    FieldSet used_fields_;
+
+    unsigned int n_dofs_;                                     ///< Number of dofs
+    unsigned int n_dofs_sub_;                                 ///< Number of dofs (on lower dim element)
+    std::vector<unsigned int> n_dofs_ngh_;                    ///< Number of dofs on lower and higher dimension element (vector of 2 items)
+    FEValues<3> fe_values_;                                   ///< FEValues of cell object (FESystem of P disc finite element type)
+    FEValues<3> fe_values_side_;                              ///< FEValues of side object
+    FEValues<3> fe_values_sub_;                               ///< FEValues of lower dimension cell object
+
+    vector<LongIdx> dof_indices_;                             ///< Vector of global DOF indices
+    vector<vector<LongIdx> > side_dof_indices_;               ///< 2 items vector of DOF indices in neighbour calculation.
+    vector<PetscScalar> local_matrix_;                        ///< Auxiliary vector for assemble methods
+    vector<vector<vector<PetscScalar>>> local_matrix_ngh_;    ///< Auxiliary vectors for assemble ngh integral
+    const FEValuesViews::Vector<3> * vec_view_;               ///< Vector view in cell integral calculation.
+    const FEValuesViews::Vector<3> * vec_view_side_;          ///< Vector view in boundary / neighbour calculation.
+    const FEValuesViews::Vector<3> * vec_view_sub_;           ///< Vector view of low dim element in neighbour calculation.
+
+    template < template<IntDim...> class DimAssembly>
+    friend class GenericAssembly;
+
+};
+
+
+template <unsigned int dim>
+class RhsAssemblyElasticity : public AssemblyBase<dim>
+{
+public:
+    typedef typename Elasticity::EqFields EqFields;
+    typedef typename Elasticity::EqData EqData;
+
+    /// Constructor.
+    RhsAssemblyElasticity(EqFields *eq_fields, EqData *eq_data)
+    : AssemblyBase<dim>(1), eq_fields_(eq_fields), eq_data_(eq_data) {
+        this->active_integrals_ = (ActiveIntegrals::bulk | ActiveIntegrals::coupling | ActiveIntegrals::boundary);
+        std::vector<string> sub_names = {"X", "d", "load", "potential_load", "cross_section", "dirichlet_penalty",
+                            "lame_mu", "lame_lambda", "young_modulus", "poisson_ratio", "bc_type", "bc_displacement",
+		                    "bc_traction", "fracture_sigma"};
+        this->used_fields_ = eq_fields_->subset( sub_names );
+    }
+
+    /// Destructor.
+    ~RhsAssemblyElasticity() {}
+
+    /// Initialize auxiliary vectors and other data members
+    void initialize(FMT_UNUSED std::shared_ptr<Balance> balance, ElementCacheMap *element_cache_map) {
```

Comment:
Can we make a common function for creation of objects shared between RHS and Stiffness assembly?